VirtoCommerce.MarketingModule.Web/Scripts/promotion/blades/promotion-detail.js

```diff
@@ -46,6 +46,30 @@ angular.module('virtoCommerce.marketingModule')
             $q.all(blade.expressionPromises).then(function (allData) {
                 blade.currentEntity = angular.copy(data);
                 blade.origEntity = data;
+
+                var conditions = _.filter(data.dynamicExpression.children, function (condition) {
```

Comment:
Can't we build productIds inside one loop?

Reply:
Fixed. Now is using one each main loop for conditions. You may watch it on the https://github.com/VirtoCommerce/vc-module-marketing/pull/63/commits/f108a5765427ee014a054be245cf8083a367d4ee page.